packages/myst-execute/src/execute.ts

```diff
@@ -281,6 +284,15 @@ export type Options = {
  */
 export async function kernelExecutionTransform(tree: GenericParent, vfile: VFile, opts: Options) {
   const log = opts.log ?? console;
+
+  // We need the kernelspec to proceed
+  if (opts.frontmatter.kernelspec === undefined) {
```

Comment:
This should never be triggered in the current CLI workflow, right? Because we always have a valid object coming out of frontmatter validation?

Reply:
If you don't define frontmatter.kernelspec, you don't get a kernelspec object: https://github.com/jupyter-book/mystmd/blob/4a5d3b789cd9c85f253593367eda2bf681b10c53/packages/myst-frontmatter/src/page/types.ts#L18